benchmarks/benchmarks/bench_linalg.py

```diff
@@ -130,4 +133,16 @@ def time_einsum_sum_mul(self, dtype):
 
     # sum and multiply:trigger sum_of_products_stride0_contig_outstride0_two
     def time_einsum_sum_mul2(self, dtype):
-        np.einsum("i...,->", self.d, 300, optimize=True)
\ No newline at end of file
+        np.einsum("i...,->", self.d, 300, optimize=True)
+    
+    # scalar mul: trigger sum_of_products_stride0_contig_outcontig_two
+    def time_einsum_mul(self, dtype):
+        np.einsum("i,->i", self.c2, 300, optimize=True)
+    
+    # trigger contig_contig_outstride0_two
+    def time_einsum_contig_contig(self, dtype):
+        np.einsum("ji,i->", self.b1, self.c3, optimize=True)
+
+    # trigger sum_of_products_contig_outstride0_one
+    def time_einsum_contig_outstride0(self, dtype):
+        np.einsum("i->", self.c2, optimize=True)
```

Comment:
Could you add some non-contiguous benchmarks

Reply:
yes, I will add non-contiguous benchmarks, The size of _multiarray_umath has increased about 73KB comparing to the master.